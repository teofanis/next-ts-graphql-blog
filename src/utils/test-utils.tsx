import { render, RenderOptions } from '@testing-library/react'
import { ReactElement } from 'react'
// Add in any providers here if necessary:

const Providers: React.FC = ({ children }) => {
  return <>{children}</>
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'
// override render method
export { customRender as render }
