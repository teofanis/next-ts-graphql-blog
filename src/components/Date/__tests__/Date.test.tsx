import { Date } from '@components/index'
import { render, within } from '@utils/test-utils'
import React from 'react'

describe('Date Test', () => {
  it('renders correctly', () => {
    const { container } = render(<Date date={'2021-10-28'} />)
    expect(container).toMatchSnapshot()
  })

  it('formats the date by default', () => {
    const date = '2021-10-28'
    const { container } = render(<Date date={date} />)
    expect(container.textContent).toBe('Oct 28, 2021')
  })

  it('formats the date by a given format', () => {
    const format = 'DD MMM YYYY'
    const date = '2021-10-28'
    const { container } = render(<Date date={date} format={format} />)
    expect(container.textContent).toBe('28 Oct 2021')
  })

  it('can render with a calendar icon', () => {
    const format = 'DD MMM YYYY'
    const date = '2021-10-28'
    const { container } = render(
      <Date date={date} format={format} withIcon={true} />
    )
    expect(container.textContent).toBe('28 Oct 2021')
    expect(container.querySelector('svg')).toBeTruthy()
    expect(within(container).getByTestId('calendar-icon')).toBeVisible()
  })
})
