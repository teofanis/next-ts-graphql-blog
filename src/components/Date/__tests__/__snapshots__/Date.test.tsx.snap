// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Date Test renders correctly 1`] = `
<div>
  <span>
    Oct 28, 2021
  </span>
</div>
`;
